Lodgify.Payments.Stripe.Server.IntegrationTests/Controllers/v1/AccountSessionControllerTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using FluentAssertions;
+using Lodgify.Payments.Stripe.Api.Models.v1.Requests;
+using Lodgify.Payments.Stripe.Api.Models.v1.Responses;
+using Lodgify.Payments.Stripe.Domain.AccountSessions;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Factories;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Mocks;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.Shared;
+using Lodgify.Payments.Stripe.Server.IntegrationTests.WireMock.Mappings;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace Lodgify.Payments.Stripe.Server.IntegrationTests.Controllers.v1;
+
+public class AccountSessionControllerTests : BaseIntegrationTest
+{
+    private const string RequestBaseUrl = "api/v1/account-sessions";
+
+    public AccountSessionControllerTests(CustomWebApplicationFactory factory) : base(factory)
+    {
+    }
+
+    [Fact]
+    public async Task Post_ShouldReturnOk_WhenAccountSessionCreated()
+    {
+        //Arrange
+        const int accountId = 2;
+        const string stripeAccountId = "acct_1Q0hEfFaeO1hhNHt";
+
+        await Insert(PredefinedMocks.AccountMock.CreateWithStripeAccountId(accountId, stripeAccountId));
+        await UseMockMapping(WiremockMappings.StripeMappings.CreateAccountSession);
+
+        //Act
+        var response = await PostAsync(accountId, RequestBaseUrl, new CreateAccountSessionRequest(stripeAccountId));
+
+        // Assert
+        response.IsSuccessStatusCode.Should().BeTrue();
+
+        var accountSessionResponse = await DeserializeResponse<CreateAccountSessionResponse>(response);
+        accountSessionResponse.Should().NotBeNull();
+        accountSessionResponse.ClientSecret.Should().NotBeNullOrEmpty();
+
+        await AssertSessionAccountCreated(accountSessionResponse.StripeAccountId, accountSessionResponse.ClientSecret);
+    }
+
+    private async Task AssertSessionAccountCreated(string stripeAccountId, string clientSecret)
+    {
+        var accountSession = await DbContext.Set<AccountSession>().Where(x => x.StripeAccountId == stripeAccountId && x.ClientSecret == clientSecret).FirstAsync();
```

Comment:
I think FirstAsync() throws exceptions if there is no element so probably better to use FirstOrDefaultAsync().
Anyway how about using Single always when we expect only 1 element to exist (not only in tests but also in the app)?

```suggestion
        var accountSession = await DbContext.Set<AccountSession>().Where(x => x.StripeAccountId == stripeAccountId && x.ClientSecret == clientSecret).SingleOrDefultAsync();
```

Reply:
The First was choose instead of FirstOrDefault to be sure there is item and not default one was return.

Reply:
But later you check if it is not null. And default is null, no?

Also why not Single?